### PR TITLE
docs(config): 添加游戏内容配置系统完整文档与验证工具实现

### PR DESCRIPTION
**Pull request by GeWuYou**
- 新增游戏内容配置系统详细文档，涵盖 YAML 配置、JSON Schema 结构、目录组织等核心概念
- 实现配置验证工具，支持运行时校验、类型检查、引用验证等功能
- 提供完整的接入模板，包括项目结构、CSProj 配置、启动帮助器等推荐实践
- 集成 VS Code 插件支持，提供配置浏览、表单编辑、递归校验等开发期工具能力
- 实现热重载机制，支持开发期配置文件变更自动刷新运行时表
- 添加详细的 Schema 示例和 YAML 示例，展示怪物、物品等游戏内容配置的实际应用
- 提供 Architecture 接入模板，支持与现有架构的无缝集成
- 实现跨表引用校验，确保配置数据的一致性和完整性

## Summary by Sourcery

在 YAML 运行时验证器、Node.js 工具链、VS Code 扩展和源码生成器中新增对 JSON Schema `const` 约束的通用支持，并在游戏配置系统文档中记录这一新行为。

New Features:
- 在 YAML 配置模式验证器中，为标量、数组和对象字段提供 JSON Schema `const` 约束支持，并在运行时强制执行和提供诊断信息。
- 在 Node.js 配置工具链中暴露并消费 `const` 元数据，包括可比较键、可编辑值和本地化校验消息，并在 VS Code 表单编辑器中以类似默认值和提示的形式呈现。
- 在生成的配置类型 XML 文档和快照输出中包含 `const` 约束信息。

Enhancements:
- 统一标量关键字归一化逻辑，使 `enum` 和 `const` 在运行时和工具链中共享一致的比较语义。
- 使用序号语义对可编辑的模式字段键和 object-const 键进行排序，以保持 C# 和 JavaScript 实现之间的一致性。
- 扩展中文游戏配置系统文档，描述共享的 `const` 支持，以及它与运行时验证、工具和热重载工作流的交互方式。

Tests:
- 新增完整的运行时测试，覆盖标量、数组和嵌套对象的 `const` 违反与匹配情况，包括空对象场景。
- 为 Node.js 工具链新增测试，用于解析、归一化和验证 `const` 元数据，以及为由 const 支持的字段生成示例 YAML 和 UI 行为。
- 更新源码生成器测试和快照，验证包括空字符串常量在内的 `const` 约束会被保留在生成的 XML 文档中。

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Add shared support for JSON Schema `const` constraints across the YAML runtime validator, Node.js tooling, VS Code extension, and source generator, and document the new behavior in the game config system docs.

New Features:
- Support JSON Schema `const` constraints for scalar, array, and object fields in the YAML config schema validator with runtime enforcement and diagnostics.
- Expose and consume `const` metadata in the Node.js config tooling, including comparable keys, editable values, and localized validation messages, and surface it in the VS Code form editor as default-like values and hints.
- Include `const` constraint information in generated configuration type XML documentation and snapshot outputs.

Enhancements:
- Unify scalar keyword normalization logic so `enum` and `const` share consistent comparison semantics in both runtime and tooling.
- Sort editable schema field keys and object-const keys using ordinal semantics to stay aligned between C# and JavaScript implementations.
- Extend Chinese game config system documentation to describe shared `const` support and how it interacts with runtime validation, tools, and hot-reload workflows.

Tests:
- Add comprehensive runtime tests covering scalar, array, and nested object `const` violations and matches, including empty-object cases.
- Add Node.js tooling tests for parsing, normalizing, and validating `const` metadata, as well as sample YAML generation and UI behavior for const-backed fields.
- Update source generator tests and snapshots to verify that `const` constraints, including empty-string constants, are preserved in generated XML documentation.

</details>

新功能：
- 在 YAML 运行时校验器中，为标量、数组和对象类型的配置字段提供 schema `const` 约束支持。
- 在 Node.js 校验/工具层中暴露 `const` 元数据（包括 UI 提示和本地化诊断信息），并按照运行时比较语义一致地强制执行。
- 在 VS Code 扩展的表单编辑器中，将 `const` 约束以默认字段值和可读性良好的提示形式呈现。
- 在由 schema 源生成的配置类型 XML 文档中包含 `const` 约束信息。

增强改进：
- 统一标量关键字归一化逻辑，使 `enum` 和 `const` 在运行时和工具中共享相同的比较语义。
- 扩展游戏配置系统文档，记录共享的 `const` 支持，以及其在运行时、工具和热重载工作流中的行为。

测试：
- 新增运行时测试，覆盖 YAML 配置中标量、数组和嵌套对象字段的 `const` 违规情况。
- 新增工具测试，确保 schema 解析能捕获 `const` 元数据，并在标量、对象和数组常量不匹配时报告校验错误。
- 更新源码生成器快照测试，以验证生成的配置文档中正确输出 `const` 约束。

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

在 YAML 运行时验证器、Node.js 工具链、VS Code 扩展和源码生成器中新增对 JSON Schema `const` 约束的通用支持，并在游戏配置系统文档中记录这一新行为。

New Features:
- 在 YAML 配置模式验证器中，为标量、数组和对象字段提供 JSON Schema `const` 约束支持，并在运行时强制执行和提供诊断信息。
- 在 Node.js 配置工具链中暴露并消费 `const` 元数据，包括可比较键、可编辑值和本地化校验消息，并在 VS Code 表单编辑器中以类似默认值和提示的形式呈现。
- 在生成的配置类型 XML 文档和快照输出中包含 `const` 约束信息。

Enhancements:
- 统一标量关键字归一化逻辑，使 `enum` 和 `const` 在运行时和工具链中共享一致的比较语义。
- 使用序号语义对可编辑的模式字段键和 object-const 键进行排序，以保持 C# 和 JavaScript 实现之间的一致性。
- 扩展中文游戏配置系统文档，描述共享的 `const` 支持，以及它与运行时验证、工具和热重载工作流的交互方式。

Tests:
- 新增完整的运行时测试，覆盖标量、数组和嵌套对象的 `const` 违反与匹配情况，包括空对象场景。
- 为 Node.js 工具链新增测试，用于解析、归一化和验证 `const` 元数据，以及为由 const 支持的字段生成示例 YAML 和 UI 行为。
- 更新源码生成器测试和快照，验证包括空字符串常量在内的 `const` 约束会被保留在生成的 XML 文档中。

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Add shared support for JSON Schema `const` constraints across the YAML runtime validator, Node.js tooling, VS Code extension, and source generator, and document the new behavior in the game config system docs.

New Features:
- Support JSON Schema `const` constraints for scalar, array, and object fields in the YAML config schema validator with runtime enforcement and diagnostics.
- Expose and consume `const` metadata in the Node.js config tooling, including comparable keys, editable values, and localized validation messages, and surface it in the VS Code form editor as default-like values and hints.
- Include `const` constraint information in generated configuration type XML documentation and snapshot outputs.

Enhancements:
- Unify scalar keyword normalization logic so `enum` and `const` share consistent comparison semantics in both runtime and tooling.
- Sort editable schema field keys and object-const keys using ordinal semantics to stay aligned between C# and JavaScript implementations.
- Extend Chinese game config system documentation to describe shared `const` support and how it interacts with runtime validation, tools, and hot-reload workflows.

Tests:
- Add comprehensive runtime tests covering scalar, array, and nested object `const` violations and matches, including empty-object cases.
- Add Node.js tooling tests for parsing, normalizing, and validating `const` metadata, as well as sample YAML generation and UI behavior for const-backed fields.
- Update source generator tests and snapshots to verify that `const` constraints, including empty-string constants, are preserved in generated XML documentation.

</details>

</details>

## Summary by CodeRabbit

* **新功能**
  * 配置系统、生成器与编辑器全面支持 JSON Schema `const`，表单提示与样例优先显示 `const` 值，生成文档/注释包含 `const` 描述。
* **错误修复 / 验证**
  * 运行时与编辑器会对 `const` 不匹配产生清晰诊断（含路径与差异）；对象比对忽略字段顺序，数组保留顺序，标量按规范化规则比较。
* **测试**
  * 大量新增单元测试覆盖标量/数组/嵌套对象的正负案例及空对象/空字符串等边界。
* **文档**
  * 更新中文文档与本地化文本，说明 `const` 语义及跨工具一致性。

### CODE REVIEW

File: GFramework.Game/Config/YamlConfigSchemaValidator.cs [hunk 1]
```diff
@@ -2576,12 +2894,51 @@ public NodeValidation(
...
+    /// <param name="comparableValue">用于与 YAML 节点比较的稳定键。</param>
+    /// <param name="displayValue">用于诊断输出的原始常量文本。</param>
+    public YamlConfigConstantValue(string comparableValue, string displayValue)
+    {
+        ArgumentException.ThrowIfNullOrWhiteSpace(comparableValue);
+        ArgumentException.ThrowIfNullOrWhiteSpace(displayValue);
+
+        ComparableValue = comparableValue;
+        DisplayValue = displayValue;
     }
```
[the remark this answers is not in the document]
Comment (GeWuYou) [reply GFramework.Game/Config/YamlConfigSchemaValidator.cs:2931]: 已更新，请检查是否解决

File: tools/gframework-config-tool/test/configValidation.test.js [hunk 2]
```diff
@@ -65,6 +65,42 @@ test("parseSchemaContent should capture nested objects and object-array metadata
...
+test("parseSchemaContent should capture const metadata for scalar, object, and array nodes", () => {
+    const schema = parseSchemaContent(`
+        {
+          "type": "object",
+          "properties": {
+            "rarity": {
+              "type": "string",
+              "const": "common"
+            },
+            "reward": {
```
[the remark this answers is not in the document]
Comment (GeWuYou) [reply tools/gframework-config-tool/test/configValidation.test.js]: 已更新，请检查
